salt/returners/local_cache.py

```diff
@@ -100,12 +100,13 @@ def prep_jid(nocache=False, passed_jid=None, recurse_count=0):
 
     # Make sure we create the jid dir, otherwise someone else is using it,
     # meaning we need a new jid.
-    try:
-        os.makedirs(jid_dir_)
-    except OSError:
-        time.sleep(0.1)
-        if passed_jid is None:
-            return prep_jid(nocache=nocache, recurse_count=recurse_count+1)
```

Comment:
The original bug is that the ``sleep()`` is not inside this conditional (``if passed_jid is None``) for just the case where a ``JID`` is generated and we need a new clock tick for a different ``JID`` to affect this recursive call.